File: dxbase/src/main/java/com/dxmovie/dxbase/binding/viewadapter/swiperefresh/ViewAdapter.java
package com.dxmovie.dxbase.binding.viewadapter.swiperefresh;


import androidx.databinding.BindingAdapter;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

import com.dxmovie.dxbase.binding.command.BindingCommand;


/**
 *
 */
public class ViewAdapter {
    //下拉刷新命令
    @BindingAdapter({"onRefreshCommand"})
    public static void onRefreshCommand(SwipeRefreshLayout swipeRefreshLayout, final BindingCommand onRefreshCommand) {
        swipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                if (onRefreshCommand != null) {
                    onRefreshCommand.executeAction();
                }
            }
        });
    }

    /**  是否可以刷新 */
    @BindingAdapter("isCanRefresh")
    public static void isSwipeRefresh(SwipeRefreshLayout swipeRefreshLayout, Boolean isRefresh) {
        swipeRefreshLayout.setEnabled(isRefresh);
    }

    /**
     * 设置正在刷新
     */
    @BindingAdapter("isRefreshing")
    public static void refreshing(SwipeRefreshLayout swipeRefreshLayout, Boolean isRefresh) {
        swipeRefreshLayout.setRefreshing(isRefresh);
    }

}

File: dxbase/src/main/java/com/dxmovie/dxbase/binding/viewadapter/swiperefresh/ViewAdapter.kt
package com.dxmovie.dxbase.binding.viewadapter.swiperefresh

import androidx.databinding.BindingAdapter
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import com.dxmovie.dxbase.binding.command.BindingCommand

class ViewAdapter {
    companion object {
        //下拉刷新命令
        @JvmStatic
        @BindingAdapter("onRefreshCommand")
        fun onRefreshCommand(swipeRefreshLayout: SwipeRefreshLayout, onRefreshCommand: BindingCommand?) {
            swipeRefreshLayout.setOnRefreshListener { onRefreshCommand?.executeAction() }
        }

        /**  是否可以刷新  */
        @JvmStatic
        @BindingAdapter("isCanRefresh")
        fun isSwipeRefresh(swipeRefreshLayout: SwipeRefreshLayout, isRefresh: Boolean?) {
            swipeRefreshLayout.isEnabled = isRefresh!!
        }

        /**
         * 设置正在刷新
         */
        @JvmStatic
        @BindingAdapter("isRefreshing")
        fun refreshing(swipeRefreshLayout: SwipeRefreshLayout, isRefresh: Boolean?) {
            swipeRefreshLayout.isRefreshing = isRefresh!!
        }
    }

}